crates/hir-ty/src/object_safety.rs

```diff
@@ -349,7 +349,7 @@ where
                 ControlFlow::Continue(())
             } else {
                 let generic_params = db.generic_params(item.into());
-                if generic_params.len_type_or_consts() > 0 {
+                if generic_params.len_type_or_consts() > 0 || generic_params.len_lifetimes() > 0 {
                     cb(ObjectSafetyViolation::GAT(it))
```

Comment:
Isn't it sufficient to use just `!generic_params.is_empty()`? Other codes LGTM

Reply:
Oh didn't check if there is such method!